src/main/resources/avro/reads.avdl

```diff
@@ -177,7 +183,41 @@ record LinearAlignment {
 }
 
 /**
-Each read alignment describes a linear alignment with additional information
+A string-to-reference-graph alignment can be represented by one CIGAR string and
+one `Path` through multiple `Reference`s, against which the CIGAR string is
+interpreted.
+
+Note that `Path`s in `GraphAlignment`s are restricted to visiting `Reference`s
+and following reference adjacencies. If a read needs to be aligned to sequences
+that are not present in a `ReferenceSet`, it needs to be aligned to a new
+`ReferenceSet` with those sequences. If a read needs to follow adjacencies that
+are not present in the `ReferenceSet` it's being aligned to, it should be
+represented as a "chimeric" alignment, and should use multiple `ReadAlignment`s
+and the supplementaryAlignment flag instead of a single `GraphAlignment`.
+
+Some especially large deletions could be represented just as well as a large
+deletion in the CIGAR string, or as a chimeric alignment.
+*/
+record GraphAlignment {
+  /** The `Path` against which the read is aligned */
+  Path path;
+
+  /**
+  The mapping quality of this alignment. Represents how likely
+  the read maps to this position as opposed to other locations.
+  */
+  union { null, int } mappingQuality = null;
+
+  /**
+  Represents the local alignment of this sequence (alignment matches, indels,
+  etc) versus the `Path`.
+  */
+  array<CigarUnit> cigar = [];
```

Comment:
I think you mean single `Segment` in a `Path`, and I'm not sure what we would gain by that. The `Segment`s in the `Path` are required to be connected by reference adjacencies, and we don't want end join adjacencies between bases to be second-class relative to on-the-same-sequence adjacencies between bases.

If I map perfectly to one end of a known translocation, I should have a CIGAR of one run of matches, and a `Path` explaining that I map to one end of a known translocation. Right?